classes/ThankYous/Api.php

```diff
@@ -592,106 +601,110 @@ public function CreateFromArray(array $data)
 
 		$errors = [];
 
-		if (isset($id))
+		if (!isset($description))
 		{
-			$thank_you = $this->GetThankYou($id);
-			if (isset($description))
-			{
-				$thank_you->SetDescription($description);
-			}
-		} else
+			$errors[] = ['name' => 'description', 'reason' => ($this->lmsg)('thankyou.thankyou.description.error.empty')];
+		} elseif (!is_string($description))
 		{
-			if (!isset($description))
-			{
-				$errors[] = ['name' => 'description', 'reason' => ($this->lmsg)('thankyou.thankyou.description.error.empty')];
-			}
+			$errors[] = ['name' => 'description', 'reason' => ($this->lmsg){'thankyou.thankyou.description.error.not_string'}];
+		}
 
-			if (!isset($author))
-			{
-				$errors[] = ['name' => 'author', 'reason' => ($this->lmsg)('thankyou.thankyou.author.error.undefined')];
-			} elseif (!is_int($author) && !($author instanceof User))
-			{
-				$errors[] = ['name' => 'author', 'reason' => ($this->lmsg)('thankyou.thankyou.author.error.invalid')];
-			}
+		if (!isset($author))
+		{
+			$errors[] = ['name' => 'author', 'reason' => ($this->lmsg)('thankyou.thankyou.author.error.undefined')];
+		} elseif (!is_int($author) && !($author instanceof User))
+		{
+			$errors[] = ['name' => 'author', 'reason' => ($this->lmsg)('thankyou.thankyou.author.error.invalid')];
+		}
 
-			if (!empty($errors))
-			{
-				throw new ValidationException($errors);
-			}
+		if (!empty($errors))
+		{
+			throw new ValidationException($errors);
+		}
 
-			$thank_you = $this->thank_you_factory->Create($author, $description, null);
+		$thank_you = $this->thank_you_factory->Create($author, $description, null);
 
-			if (isset($id) && !is_int($id))
+		if (isset($thanked))
+		{
+			$this->SetThankedFromArray($thank_you, $thanked);
+		}
+
+		if (isset($tag_ids))
+		{
+			$this->SetTagsFromArray($thank_you, $tag_ids);
+		}
+
+		if (!empty($errors))
+		{
+			throw new ValidationException($errors);
+		}
+
+		$this->PopulateThankYouUsersFromThankables($thank_you);
+
+		return $this->SaveNew($thank_you);
+	}
+
+	/**
+	 * @param SecurityContext $context
+	 * @param int             $id
+	 * @param array           $data
+	 * @return int
+	 * @throws TagNotFoundException
+	 * @throws ThankYouForbiddenException
+	 * @throws ThankYouNotFoundException
+	 * @throws ValidationException
+	 */
+	public function UpdateAndSave(SecurityContext $context, int $id, array $data): int
+	{
+		$description = $data['description'] ?? null;
+
+		$thanked = $data['thanked'] ?? null;
+		$tag_ids = $data['tags'] ?? null;
+
+		$errors = [];
+
+		$thank_you = $this->GetThankYou($id);
+
+		if (isset($description))
+		{
+			if (!is_string($description))
 			{
-				$errors[] = ['name' => 'id', 'reason' => ($this->lmsg)('thankyou.thankyou.id.error.invalid')];
+				$errors[] = ['name' => 'description', 'reason' => ($this->lmsg){'thankyou.thankyou.description.error.not_string'}];
```

Comment:
Looks like an accidental change of brackets :slightly_smiling_face:

Reply:
Sorry my fault! Rushing 🙄.
Fixed in 4039c99708276510d9873b1f099af0c77a5d07f0